src/main/java/nuffle/ui/Ui.java

```diff
@@ -0,0 +1,141 @@
+package nuffle.ui;
+
+import nuffle.exception.NuffleException;
+import nuffle.task.Task;
+import java.util.ArrayList;
+
+/**
+ * The Ui class handles the interaction with the user.
+ * It displays messages to the user based on the commands executed.
+ */
+public class Ui {
+
+    /**
+     * Prints a line separator to the console.
+     */
+    private static void printLine() {
+        System.out.println("---------------------------------------------");
+    }
+
+    /**
+     * Displays the welcome message when the application starts.
+     */
+    public static void welcomeMessage() {
```

Comment:
perhaps the method's name could be `printWelcomeMessage` instead, which is phrased to sound more like an action than a noun